### DISCUSSION TITLE
How to monitor Hetnzer Ubuntu VM

### DISCUSSION THREAD

**Discussion by MohdRashid01** [community/community#45574] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

My 4 VMs are running in Hetzner Cloud

I have monitor 4 VMs of ubuntu OS running on it and some data is also running in all 4 VMs.

I want to write a shell script on that 4 VMs whether they are Up/Down like ping

That script should check daily or every hour about 4 VMs is Up/Down or Not

Please give me some link or tips how can i write it

and Shell script i will keep in jenkins job to daily it will check

**Comment by jge162** [comment, marked as answer]
Hello @MohdRashid01 Thank you for the question, although it is not Github related. Here is a potential solution. 
You can use the nmap command to scan the VMs and determine if they are up or down. You can use the command nmap -sn <IP Address or Range> to scan the IP addresses of the VMs and then use an if/else statement to print a result of whether they are up or down. You can then save the script and set up a Jenkins job to run the script every hour. 

[Ubuntu Jenkins guide](https://www.digitalocean.com/community/tutorials/how-to-install-jenkins-on-ubuntu-16-04)

**Reply by MohdRashid01** [reply]
@jge162  Thanks you for much i will implement this
